components/layout_thread/lib.rs

```diff
@@ -954,18 +954,23 @@ impl LayoutThread {
             }
             let display_list = (*rw_data.display_list.as_ref().unwrap()).clone();
 
+            let mut iframe_sizes = Vec::new();
             for item in &display_list.list {
                 match *item {
                     DisplayItem::Iframe(ref iframe) => {
                         let size = Size2D::new((*item).bounds().size.width.to_f32_px(),
                                                (*item).bounds().size.height.to_f32_px());
                         debug!("new iframe size: {:?}", size);
-                        let msg = ConstellationMsg::FrameSize((*iframe).iframe, size);
-                        if let Err(e) = self.constellation_chan.send(msg) {
-                            warn!("Layout resize to constellation failed ({}).", e);
-                        }
+                        iframe_sizes.push(((*iframe).iframe, size));
```

Comment:
This is still walking all the display list instead of doing it at display list building time. You should collect the sizes in `build_fragment_type_specific_display_items` in `components/layout/display_list_builder.rs`, under the `Iframe` branch.

For that, you probably want to add the vector of iframe sizes to `DisplayListBuildState` (in that same file), and send the sizes to the constellation once you're done building the display list.